File: php/difference-of-squares/difference-of-squares.php
<?php
  function difference($number){
      $squareOfSums = squareOfSums($number);
      $sumOfSquares = sumOfSquares($number);
      $difference = $squareOfSums - $sumOfSquares;
      return $difference;
    }
  function squareOfSums($number){
    $sum = 0;
    for($i = 0; $i < ($number + 1); $i++){
      $sum += $i;
    }
    $squaredSum = $sum * $sum;
    return $squaredSum;
  }
  function sumOfSquares($number){
    $sumSquared = 0;
    for($j = 0; $j < ($number + 1); $j++){
      $square = $j * $j;
      $sumSquared += $square;
    }
    return $sumSquared;
  }
?>

File: ruby/difference-of-squares/difference_of_squares.rb
module BookKeeping
  VERSION = 3
end

class Squares
  def initialize(number)
    @number = number
  end

  def sum_of_squares
    sum_squares = 0
    (1..@number).each do |i|
      sum_squares += i ** 2
    end
    sum_squares
  end

  def square_of_sum
    sum = 0
    (1..@number).each do |i|
      sum += i
    end
    sum * sum
  end

  def difference
    self.square_of_sum - self.sum_of_squares
  end
end
